src/main/java/de/thomas_oster/liblasercut/drivers/GenericGcodeDriver.java

```diff
@@ -549,8 +556,23 @@ protected void http_upload(URI url, String data, String filename) throws IOExcep
 
   protected void http_play(String filename) throws IOException, URISyntaxException
   {
+    String command = "play /sd/"+filename;
+    http_command(command);
+  }
+
+  protected void http_commands(String commands, String filename) throws IOException, URISyntaxException
+  {
+    for (String command : commands.split(","))
+    {
+      command = command.replace("$filename", filename);
```

Comment:
Looking at this, I wondered whether this should not be `$jobname`, e.g. like used [here](https://github.com/t-oster/VisiCut/blob/b498e3a129e1cff4aa3657bd93751b10a52d1d13/src/main/java/de/thomas_oster/visicut/gui/MainView.java#L2321). However, it seems that these are actually slightly different values, since the liblasercut adds the `.gcode` extension, transforming the the jobname into a filename [here](https://github.com/t-oster/LibLaserCut/blob/d6728e8df802d631fd9f1e54d73fa45a2a26d04b/src/main/java/de/thomas_oster/liblasercut/drivers/GenericGcodeDriver.java#L823).

IOW, using `$filename` here seems good to me.

Reply:
That's exactly my reasoning why i finally came up with $filename :)